server/http/interceptors/interceptors_test.go

```diff
@@ -0,0 +1,107 @@
+package interceptors
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRedirectIfNotForwardedHTTPS(t *testing.T) {
+	originalVal := *upgradeInsecure
+	t.Cleanup(func() {
+		*upgradeInsecure = originalVal
+	})
+	*upgradeInsecure = true
+
+	tests := []struct {
+		name            string
+		route           string
+		expectedCode    int
+		expectedHeaders http.Header
+		sslServerAddr   string
+		setup           func(*http.Request)
+	}{
+		{
+			"https request",
+			"/foo",
+			http.StatusOK,
+			http.Header{},
+			"example.com",
+			func(req *http.Request) {
+				req.Header.Set("X-Forwarded-Proto", "https")
+			},
+		},
+		{
+			"http request with X-Forwarded-Proto header",
+			"/foo",
+			http.StatusMovedPermanently,
+			http.Header{
+				"Location": []string{"https://example.com/foo"},
+			},
+			"example.com",
+			func(req *http.Request) {
+				req.Header.Set("X-Forwarded-Proto", "http")
+			},
+		},
+		{
+			"http request without X-Forwarded-Proto header",
+			"/foo",
+			http.StatusMovedPermanently,
+			http.Header{
+				"Location": []string{"https://example.com/foo"},
+			},
+			"example.com",
+			func(req *http.Request) {
+				req.Header.Del("X-Forwarded-Proto")
+			},
+		},
+		{
+			"healthcheck request without X-Forwarded-Proto header",
+			"/health",
+			http.StatusOK,
+			http.Header{},
+			"example.com",
+			func(req *http.Request) {
+				req.Header.Del("X-Forwarded-Proto")
+				req.Header.Set("User-Agent", "GoogleHC/1.0")
+			},
+		},
+		{
+			"custom ssl address",
```

Comment:
Use named fields so it's more obvious what each of these values mean w/o having to look back at the struct definition.